guide/additional-info/async-await.md

```diff
@@ -0,0 +1,158 @@
+
+A Promise can have 3 states:
+
+* pending
+* fulfilled
+* rejected
+
+The state, **pending**, means that the Promise still is ongoing and nether fulfilled nor rejected.
+The state, **fulfilled**, means that the Promise is done and was executed without any errors.
+The state, **rejected**, means that the Promise encountered an error and could not be executed correctly.
+
+One important thing to know is that a Promise can only have 1 state; it can never be pending and fulfilled, rejected and fulfilled, or pending and rejected. You may be asking, "How would that look in code?". Here is a small example:
+(ES6 code is being used in this example; if you do not know what that is, you should read up on that [here](/additional-info/es6-syntax).)
+
+```js
+function asyncTask() {
+	return Promise((resolve, reject) => {
+		setTimeout(resolve("Task is done"), 2000)
+	})
+}
+
+asyncTask().then(value => {
+	// asyncTask is complete and has not encountered any errors
+	// The resolved value will be the String "Task is done"
+}).catch(error => {
+	// asyncTask encountered an error
+	// the error will be an Error Object
+})
+```
+
+In this scenario the asyncTask returns a Promise and we attach a .then() function and a .catch() function to it. The .then() function will trigger if the Promise was fulfilled and the .catch() function if the Promise was rejected. But with our function we resolve the Promise after 2 seconds with the String "Task is Done", so the .catch() function will never be executed.
+
+## How to implement async/await
+
+After knowing how Promises works and what they are for, let's look at an example in which we handle multiple Promises. Let's say you wanted to react to a message in a certain order with letters (regional indicators). For this example, I will take the basic template for a discord.js bot with some es6 adjustments.
+
+```js
+const Discord = require('discord.js');
+const client = new Discord.Client();
+
+const prefix = '?';
+
+client.on('ready', () => {
+	console.log('I am ready!');
+});
+
+client.on('message', message => {
+	if (message.content === `${prefix}react`) {
+		//code inside here
+	}
+});
+
+client.login('tokeninhere');
+```
+
+So now we need to put the code in. If you don't know how node.js asynchronous execution works, you would probably try something like this:
+
+```js
+client.on('message', message => {
+	if (message.content === `${prefix}react`) {
+		message.react('🇦');
+		message.react('🇧');
+		message.react('🇨');
+	}
+});
+```
+
+But since all of these react methods are started at the same time, it would just be a race which server request finished first, so there would be no guarantee that it would react in the order you wanted it to. In order to make sure it reacts in order (a, b, c), we need to use the .then() callback from the Promises that these methods return. As a result the code we want would mostly look like this:
```

Comment:
* "... a race which server request finished first ..." -> "... a race to which server request finished first ...".